src/StudioCore/AssetUtils.cs

```diff
@@ -0,0 +1,216 @@
+﻿using SoulsFormats;
+using StudioCore.Editor;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace StudioCore;
+
+/// <summary>
+///     Helper functions and statics for Project/Assetlocator
+/// </summary>
+public class AssetUtils
+{
+    public static readonly string GameExecutableFilter;
+    public static readonly string ProjectJsonFilter;
+    public static readonly string RegulationBinFilter;
+    public static readonly string Data0Filter;
+    public static readonly string ParamBndDcxFilter;
+    public static readonly string ParamBndFilter;
+    public static readonly string EncRegulationFilter;
+    public static readonly string ParamLooseFilter;
+    public static readonly string CsvFilter;
+    public static readonly string TxtFilter;
+    public static readonly string FmgJsonFilter;
+
+    static AssetUtils()
+    {
+        // These patterns are meant to be passed directly into PlatformUtils.
+        // Everything about their handling should be done there.
+
+        // Game Executable (.EXE, EBOOT.BIN)|*.EXE*;*EBOOT.BIN*
+        // Windows executable (*.EXE)|*.EXE*
+        // Playstation executable (*.BIN)|*.BIN*
+        GameExecutableFilter = "exe,bin";
+        // Project file (project.json)|PROJECT.JSON
+        ProjectJsonFilter = "json";
+        // Regulation file (regulation.bin)|REGULATION.BIN
+        RegulationBinFilter = "bin";
+        // Data file (Data0.bdt)|DATA0.BDT
+        Data0Filter = "bdt";
+        // ParamBndDcx (gameparam.parambnd.dcx)|GAMEPARAM.PARAMBND.DCX
+        ParamBndDcxFilter = "parambnd.dcx";
+        // ParamBnd (gameparam.parambnd)|GAMEPARAM.PARAMBND
+        ParamBndFilter = "parambnd";
+        // Enc_RegBndDcx (enc_regulation.bnd.dcx)|ENC_REGULATION.BND.DCX
+        EncRegulationFilter = "bnd.dcx";
+        // Loose Param file (*.Param)|*.Param
+        ParamLooseFilter = "param";
+        // CSV file (*.csv)|*.csv
+        CsvFilter = "csv";
+        // Text file (*.txt)|*.txt
+        TxtFilter = "txt";
+        // Exported FMGs (*.fmg.json)|*.fmg.json
+        FmgJsonFilter = "fmg.json";
+        // All file filter is implicitly added by NFD. Ideally this is used explicitly.
+        // All files|*.*
+    }
+    public static GameType GetGameTypeForExePath(string exePath)
+    {
+        var type = GameType.Undefined;
+        if (exePath.ToLower().Contains("darksouls.exe"))
```

Comment:
It might, but then there's the nuanced cases like des vs bb, or in other such methods the shared cases.
For the whole file, I think ideally we'd define an object/jumptable and do some scummy OOP to handle per-game behaviours (and reuse this "game implementation" class for different databanks eg param loading code, fmg loading code etc.)
But moving to that sort of system might be a step beyond what this PR is focusing on, which is more to instantiate projects and databanks separately, as well as just clean up some asset lookup code.

That is, if I'm understanding your point.